docs/user-guide/kubeadm-reference.md

```diff
@@ -0,0 +1,126 @@
+# kubeadm reference
+
+This document specifies the advanced options of kubeadm.
+
+When run, `kubeadm init` will bootstrap a Kubernetes cluster. This is made up
+of a few steps, namely:
+
+1. Generating a token that additional nodes can use to register themselves with
+the master in future.
+
+1. Generating a self-signed CA using openssl in order to provision identities
+for each node in the cluster, and for the API server to secure communication
+with clients.
+
+1. Outputting a kubeconfig file for the kubelet to use to connect to the API server,
+as well as an additional kubeconfig file for administration.
+
+1. Generating Kubernetes resource manifests for the API server, controller manager
+and scheduler, and placing them in `/etc/kubernetes/manifests`. The kubelet watches
+this directory for static resources to create on startup. These are the core
+components of Kubernetes, and once they are up and running we can use `kubectl`
+to set up/manage any additional components.
+
+1. Any addon components (eg. DNS, discovery) are installed in the aformentioned manner,
+through the API server.
+
+## Usage
+
+Fields that support multiple values will do so either with comma
+separation, or by specifying the flag multiple times.
+
+### `kubeadm init`
+
+It is usually sufficient to run `kubeadm init` without any flags,
+but in some cases you might like to override the default behaviour.
+Here we specify all the flags that can be used to customise the Kubernetes
+installation.
+
+- `--api-advertise-addresses` (multiple values are allowed)
+- `--api-external-dns-names` (multiple values are allowed)
+
+By default, `kubeadm init` will automatically detect IP addresses and use
+these to generate certificates for the API server. This uses the IP address
+of the default network interface. If you would like to access the API server
+through a different IP address, or through a hostname, you can override these
+defaults with `--api-advertise-addresses` and `--api-external-dns-names`.
+Specifying `--api-advertise-addresses` will disable auto detection.
+
+- `--cloud-provider`
+
+Currently, `kubeadm init` does not provide autodetection of cloud provider.
+This means that load balancing and persistent volumes are not supported out
+of the box. You can specify a cloud provider using `--cloud-provider`.
+Valid values are the ones supported by `controller-manager`, namely `"aws"`,
+`"azure"`, `"cloudstack"`, `"gce"`, `"mesos"`, `"openstack"`, `"ovirt"`,
+`"rackspace"`, `"vsphere"`. In order to provide additional configuration for
+the cloud provider, you should create a `/etc/kubernetes/cloud-config.json`
+file manually, before running `kubeadm init`. `kubeadm` will then automatically
+pick those settings up and ensure other nodes are configured correctly.
```

Comment:
Say that if the user specifies this, he also has to make the corresponding changes in `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` for it to work
(`--cloud-provider` and `--cloud-config` should be set in that case)